src/workers/python/init.py.ts

```diff
@@ -1,44 +1,116 @@
+export const INITIALIZE_PYTHON_BACKEND = "__init_papyros";
+export const PROCESS_PYTHON_CODE = "__process_code";
 export const INITIALIZATION_CODE =
     `
-from pyodide import to_js
+from pyodide import to_js, eval_code_async
+from js import console
 import sys
+import ast
+import json
+import html
+import types
+import os
+import micropip
+await micropip.install('friendly_traceback')
+import friendly_traceback
+from friendly_traceback.core import FriendlyTraceback
 
-def __override_builtins(cb):
-    __capture_stdout(cb)
-    __override_stdin(cb)
-    # set name to main instead of builtins
-    globals()["__name__"] = "__main__"
-
-def __capture_stdout(cb):
-    class _OutputWriter:
-        def __init__(self):
-            self.encoding = "utf-8"
-            
-        def write(self, s):
-            cb(to_js({"type": "output", "data":s}))
-
-        def flush(self):
-            pass # Given data is always immediately written
-
-    sys.stdout = _OutputWriter()
-
-__papyros_input = ""
-def __override_stdin(cb):
-
-    def __papyros_input(prompt=""):
-        return __papyros_readline(prompt=prompt)[:-1] # Remove newline
-
-    def __papyros_readline(n=-1, prompt=""):
-        global __papyros_input
-        if not __papyros_input:
-            __papyros_input = cb(to_js({"type": "input", "data": prompt})) + "\\n"
-        if n < 0 or n > len(__papyros_input):
-            n = len(__papyros_input)
-        to_return = __papyros_input[0:n]
-        __papyros_input = __papyros_input[n:]
+__papyros = None
+
+class __Papyros():
+    def __init__(self, cb):
+        self.cb = cb
+        self.line = ""
+        self.override_builtins()
+
+    def message(self, data):
+        return self.cb(to_js(data))
+
+    def override_builtins(self):
+        self.override_output()
+        self.override_input()
+
+    def override_output(self):
+        class __OutputWriter:
+            def __init__(self, type, on_write):
+                self.encoding = "utf-8"
+                self.type = type
+                self.on_write = on_write
+                
+            def write(self, s):
+                self.on_write(dict(type=self.type, data=s))
```

Comment:
```suggestion
                if isinstance(s, bytes):
                    s = s.decode("utf8", "replace")
                self.on_write(dict(type=self.type, data=s))
```

Some libraries (e.g. flask) write bytes directly to stdout/stderr.